### ISSUE TITLE
Groupvars not working when executing playbook from subdirectory (2.3)

### ISSUE THREAD

**Issue by rawbertp** [ansible/ansible#23692]
##### ISSUE TYPE

 - Bug Report

##### ANSIBLE VERSION

```
ansible 2.3.0.0 (detached HEAD d56ba0945d) last updated 2017/04/13 08:27:38 (GMT +200)
```

##### SUMMARY

When executing a playbook from a subdirectory the group vars are not picked up (correctly). This works just fine with 2.1.

##### STEPS TO REPRODUCE

./test.yml and ./subdir/test.yml:
```yaml
---
- hosts: testhosts

  tasks:
    - debug: var=test123
```
./group_vars/all.yml:
```yaml
---
test123: test
```
./inventory/test:
```
[servers]
server1     ansible_ssh_host=127.0.0.1 ansible_connection=ssh
...
[testhosts]
server1
```

(1) Executed from the Ansible root directory (works):
```
$ ansible-playbook -i inventory/test test.yml
```

(2) Executed from a subdirectory (doesn't work):
```
$ ansible-playbook -i inventory/test subdir/test.yml
```

##### EXPECTED RESULTS

```
TASK [debug] ****************************************************************************************************************************************************************************************************
ok: [server1] => {
    "changed": false, 
    "test123": "test"
}
```

##### ACTUAL RESULTS

```
TASK [debug] ****************************************************************************************************************************************************************************************************
ok: [server1] => {
    "changed": false, 
    "test123": "VARIABLE IS NOT DEFINED!"
}
```

**Comment by abadger** [comment]
There was a bug in earlier versions of Ansible which found groupvars and hostvars located in the current working directory.  That bug has been fixed in 2.3.  groupvars and hostvars directories may be located next to the inventory or next to the playbook: https://docs.ansible.com/ansible/intro_inventory.html#splitting-out-host-and-group-specific-data  So in your example, they should live in inventory/group_vars/all.yml or in subdir/group_vars/all.yml

The reason that current working directory is not supposed to be used is that it can lead to unexpected situations and can lead to security problems for the unwary.  For instance, if someone executes ansible-playbook from /var/tmp, anyone could inject groupvars and hostvars into that run of the command.

If you'd like to discuss this more, feel free to drop by the mailing list or IRC channel.  I'll paste information about those below.

List Information
================

Hi!

Thanks very much for your interest in Ansible.  It sincerely means a lot to us. 

This appears to be a user question, and we'd like to direct these kinds of things to either the mailing list or the IRC channel.

   * IRC: #ansible on irc.freenode.net   
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project

If you can stop by there, we'd appreciate it.  This allows us to keep the issue tracker for bugs, pull requests, RFEs and the like.

Thank you once again and we look forward to seeing you on the list or IRC.  Thanks!

**Comment by Galser** [comment]
Same problem here  
ansible 2.3.2.0, 
python version = 2.7.10 (default, Oct 23 2015, 19:19:21) [GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.5)]

I have a beautiful top structure (ad recommended in official manual)  : 
...
_dev ( environment inventory )
group_vars -> all.yml
host_vars
library
..
playbooks
prod
roles_
....
Where in group_vars/all.yml   I have , for example,  variable  : 
**client_max_body_size : "20M"**

So if I run the playbook like  : 
**ansible-playbook -i dev playbooks/nginx-lb.yml**  -it fails, and cannot find the variable : 
`FAILED! => {"changed": false, "failed": true, "msg": "AnsibleUndefinedVariable: 'client_max_body_size' is undefined"} `
if i run it directly from the top folder : 
**ansible-playbook -i dev nginx-lb.yml** 
Everything work fine.
